master/internal/experiment/bulk_action.go

```diff
@@ -798,6 +798,20 @@ func MoveExperiments(
 			return nil, fmt.Errorf("updating run's project IDs: %w", err)
 		}
 
+		if _, err = tx.NewRaw(`UPDATE runs SET local_id=s.local_id FROM runs as r 
+		JOIN (SELECT r.id as id, (p.max_local_id + ROW_NUMBER() OVER(PARTITION BY p.id)) as local_id
+		FROM projects p JOIN runs r ON r.project_id=p.id) as s ON r.id=s.id WHERE r.experiment_id IN (?)`,
```

Comment:
Couple things:

1. Nit: when queries get complicated and especially when they're highly nested like this, I tend to find it worth it to lay them out so the nesting is obvious from the indentation. It helps my brain grok what I'm looking at a lot easier.
2. Possible issue worth talking about before approval: I may be remembering wrong, but isn't the order within a window undefined unless there's a `ORDER BY` clause? If so, I think it's worth adding an `ORDER BY` here.

```suggestion
		if _, err = tx.NewRaw(`
		UPDATE runs SET local_id=s.local_id
		FROM
			runs as r 
			JOIN (
				SELECT
					r.id as id,
					(p.max_local_id + ROW_NUMBER() OVER(PARTITION BY p.id ORDER BY p.id)) as local_id
				FROM
					projects p
					JOIN runs r ON r.project_id=p.id
			) as s ON r.id=s.id
		WHERE r.experiment_id IN (?)`,
```

Reply:
I'm aware the partitioning happens before the `ORDER BY`. The point I'm making is that *within each partition*, the order of the rows (which is what `ROW_NUMBER()` depends on) is not defined unless an `ORDER BY` is specified. So if there are, say ten runs associated with a given project, each will have an arbitrary `local_id` (and possibly different ones each time you run the query, depending on how Postgres decides to handle it) unless I'm mistaken. Or am I just totally misunderstanding Runs or something?

Reply:
So I was looking over the query and I realised that it was actually incorrect to begin with. As it was setting the local id of all the runs in the project rather than just the ones that were moved. 
Also I misread your original comment and didn't see the `ORDER BY` within the partition. You're right that would be necessary, thanks.